build.gradle.kts

```diff
@@ -27,16 +27,8 @@ val yaml = Yaml(DumperOptions().also {
     it.defaultFlowStyle = DumperOptions.FlowStyle.BLOCK
 })
 
-val FRONTMATTER_REGEX = Regex("---\n(.*)\n---", RegexOption.DOT_MATCHES_ALL)
-
-fun extractFrontMatter(file: File): MutableMap<String, Any?> {
-    val content = FRONTMATTER_REGEX.find(file.readText())
-    val yamlContent = content?.groupValues?.get(1) ?: error("Can't find frontmatter ('---')")
-    return yaml.load(yamlContent) as MutableMap<String, Any?>
-}
-
 fun updateFrontMatter(file: File, data: Map<String, Any?>) {
-    file.writeText(FRONTMATTER_REGEX.replace(file.readText(), "---\n" + yaml.dump(data).trim() + "\n---"))
+    file.writeText("---\n" + yaml.dump(data).trim() + "\n---")
```

Comment:
The original code was updating the front matter, while preserving the content of the file (after the `---...---` front mattert hing).
This code seems to be wrong since it is replacing the file with just the front matter without preserving the content.